openc3/lib/openc3/microservices/reducer_microservice.rb

```diff
@@ -155,50 +170,78 @@ def shutdown
 
     def metric(name)
       start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
-      yield
+      processed = yield
       elapsed = Process.clock_gettime(Process::CLOCK_MONOTONIC) - start # seconds as a float
-      @metric.set(name: name, value: elapsed, type: 'gauge', unit: 'seconds')
+      if processed
+        sample_name = name + '_sample_seconds'
+        @metric.set(name: sample_name, value: elapsed, type: 'gauge', unit: 'seconds')
+        max_name = name + '_max_seconds'
+        previous_max = @previous_metrics[max_name] || 0.0
+        if elapsed > previous_max
+          @metric.set(name: max_name, value: elapsed, type: 'gauge', unit: 'seconds')
+          @previous_metrics[max_name] = elapsed
+        end
+      end
     end
 
     def reduce_minute
-      metric(MINUTE_METRIC) do
-        ReducerModel
-          .all_files(type: :DECOM, target: @target_name, scope: @scope)
-          .each do |file|
-            process_file(file, 'minute', MINUTE_ENTRY_NSECS, MINUTE_FILE_NSECS)
-            ReducerModel.rm_file(file)
+      @mutex.synchronize do
+        metric(MINUTE_METRIC) do
+          processed = false
+          ReducerModel
+            .all_files(type: :DECOM, target: @target_name, scope: @scope)
+            .each do |file|
+              process_file(file, 'minute', MINUTE_ENTRY_NSECS, MINUTE_FILE_NSECS)
+              ReducerModel.rm_file(file)
+              processed = true
           end
+          processed # return to yield
```

Comment:
So the only way this stays false is if nothing is returned from `all_files`?

Reply:
Correct.  It only updates the metric if work was done.  Not if it just realized there was nothing to do.   Removes the more common super small time metrics.